### DISCUSSION TITLE
AsciiDoc rendered preview

### DISCUSSION THREAD

**Discussion by jansol** [zed-industries/zed#26082] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

There is an extension for AsciiDoc syntax highlighting, but previewing is not supported and AFAIK can currently not be supported with extensions.

In terms of features AsciiDoc is fairly close to Markdown so I expect this to be quite straightforward to implement piggybacking on top of the existing Markdown previews.

**Comment by pi3ch** [comment]
asciidoctor is markdown on steroid. highly needed!

**Comment by dunyakirkali** [comment]
I am the current maintainer of the zed-asciidoc extension. 

If someone from the Zed team could give me some guidance, I could look into implementing the preview function for it.

**Reply by SjoerdV** [reply]
Not a member of the Zed team but... the project that needs to be integrated is this one: https://github.com/asciidoctor/asciidoctor.js/

Similar to how the VSCode extension does it: https://github.com/asciidoctor/asciidoctor-vscode

(EDIT: JVM and Ruby based parsers are also available with Asciidoctor: https://github.com/asciidoctor/asciidoctor)

**Reply by pi3ch** [reply]
@dunyakirkali did you have any progress on this?
